packages/extension/src/serviceWorker/index.ts

```diff
@@ -293,7 +297,288 @@ chrome.runtime.onInstalled.addListener(async (details) => {
       await chrome.storage.sync.clear();
       await chrome.storage.sync.set({
         allowedNumberOfTabs: 'single',
+        isUsingCDP: true,
       });
     }
   });
 });
+
+chrome.debugger.onEvent.addListener(async (source, method, params) => {
+  // eslint-disable-next-line complexity
```

Comment:
I think we are doing too much in this listener, let's refactor the code and simplify.

Reply:
Need to do that because we want to handle both single tab and multitab condition.
Will look into this what I can do to make it more simpler.

Reply:
Refactored the whole event handler and added some utility functions too for simplicity and better code readability.